vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
+            affected_packages.append(
+                AffectedPackage(
+                    package=PackageURL(
+                        name=pkg_name, namespace=ns, type=get_type_from_ecosystem(ecosystem)
+                    ),
+                    affected_version_range=affected_range,
+                )
+            )
+        advisory = get_item(node, "advisory")
+        if not advisory:
+            LOGGER.error(f"No advisory found in {node!r}")
+            continue
+        references = get_item(advisory, "references") or []
+        if references:
+            references: List[Reference] = list(extract_references(references))
+        vuln_desc = get_item(advisory, "summary")
+        identifiers = get_item(advisory, "identifiers") or []
+        for identifier in identifiers:
+            # collect CVEs
+            if identifier["type"] == "CVEjson_object":
+                cve_ids.add(identifier["value"])
 
-                        unaffected_purls = [
-                            PackageURL(name=pkg_name, namespace=ns, version=version, type=pkg_type)
-                            for version in unaff_vers
-                        ]
-                    cve_ids = set()
-                    references = self.extract_references(adv["node"]["advisory"]["references"])
-                    vuln_desc = adv["node"]["advisory"]["summary"]
-
-                    for identifier in adv["node"]["advisory"]["identifiers"]:
-                        # collect CVEs
-                        if identifier["type"] == "CVE":
-                            cve_ids.add(identifier["value"])
-
-                        # attach the GHSA with severity score
-                        if identifier["type"] == "GHSA":
-                            for ref in references:
-                                if ref.reference_id == identifier["value"]:
-                                    ref.severities = [
-                                        VulnerabilitySeverity(
-                                            system=scoring_systems["cvssv3.1_qr"],
-                                            value=adv["node"]["advisory"]["severity"],
-                                        )
-                                    ]
-                                    # Each Node has only one GHSA, hence exit after attaching
-                                    # score to this GHSA
-                                    break
-
-                    for cve_id in cve_ids:
-                        adv_list.append(
-                            Advisory(
-                                vulnerability_id=cve_id,
-                                summary=vuln_desc,
-                                affected_packages=nearest_patched_package(
-                                    affected_purls, unaffected_purls
-                                ),
-                                references=references,
+            # attach the GHSA with severity score
+            if identifier["type"] == "GHSA":
```

Comment:
IMHO use elif here:
```suggestion
            elif identifier["type"] == "GHSA":
```

then add an else: afterwards... if the id is neither GHSA nor CVE and we do not handle it yet it must be logged as an error.